File: 7.reverse-integer.js


/*
 * @lc app=leetcode id=7 lang=javascript
 *
 * [7] Reverse Integer
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    const sign = Math.sign(x);

    x = parseInt(x.toString().split("").map(i => parseInt(i)).reverse().join(""));
    x = Math.abs(x) * sign;

    if (x > (Math.pow(2, 31) - 1) || x < -(Math.pow(2, 31))){
        return 0;
    }

    return x;
};
// @lc code=end

File: 7.reverse-integer.cs
/*
 * @lc app=leetcode id=7 lang=csharp
 *
 * [7] Reverse Integer
 */

// @lc code=start
public class Solution
{
    public int Reverse(int x)
    {
        int[] digits = x.ToString().Select(i => Convert.ToInt32(i)).ToArray();
        int[] tempDigits = digits;
        
        digits.Reverse(tempDigits);

        return digits;
    }
}
// @lc code=end

